posthog/api/authentication.py

```diff
@@ -247,16 +248,22 @@ def create(self, request: Request, *args: Any, **kwargs: Any) -> Any:
             )
 
         with transaction.atomic():
-            device = default_device(user)
-            is_allowed = device.verify_is_allowed()
-            if not is_allowed[0]:
-                raise serializers.ValidationError(detail="Too many attempts.", code="2fa_too_many_attempts")
-            if device.verify_token(request.data["token"]):
-                return self._token_is_valid(request, user, device)
-
-        # Failed attempt so increase throttle
-        device.throttle_increment()
-        raise serializers.ValidationError(detail="2FA token was not valid", code="2fa_invalid")
+            # Try TOTP device first
+            totp_device = default_device(user)
+            if totp_device:
+                is_allowed = totp_device.verify_is_allowed()
+                if not is_allowed[0]:
+                    raise serializers.ValidationError(detail="Too many attempts.", code="2fa_too_many_attempts")
+                if totp_device.verify_token(request.data["token"]):
+                    return self._token_is_valid(request, user, totp_device)
+                totp_device.throttle_increment()
+
+            # Then try backup codes
```

Comment:
Need to write tests for this change

Reply:
One thing I'm not sure about - I think normally using a backup code is a one time operation... Doing it should require them to re-setup 2fa right? 

Currently it doesn't look like the code is marked as used unless there is some under-the-hood thing I've missed? Appreciate this widens the scope but it feels potentially important 🤔

Reply:
Yeah, after going to bed I thought I should pull that change out into another PR and spend more time on it anyways. 
 I'm going to remove that change here and do it on another PR + look into your comment about resetting the 2FA.